geo/geomfn: use GEOSRelatePattern for ST_ContainsProperly

otan opened the pull request:
GEOSRelatePattern seems to avoid the checks for wildcards, making
ST_ContainsProperly faster.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49759)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49759#-:-M8m-ywZ7WSiKefhfw3h:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geomfn/de9im.go, line 28 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49759#-M8m-po-CDfCjmrq_KeD:-M8m-po0AWHx7uyZnbAw:bxl0ka) ([raw file](https://github.com/cockroachdb/cockroach/blob/4aa4b430140d4beb22c15da0c924bac3fbccae36/pkg/geo/geomfn/de9im.go#L28)):*
> ```Go
> }
> 
> // RelatePattern returns the whether the DE-9IM relation between A and B matches.
> ```

... returns whether the ... matches the pattern.


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49759)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/geomfn/de9im.go, line 28 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49759#-M8m-po-CDfCjmrq_KeD:-M8meUNK4zn7B12VznMR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4aa4b430140d4beb22c15da0c924bac3fbccae36/pkg/geo/geomfn/de9im.go#L28)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

... returns whether the ... matches the pattern.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan committed:
geo/geomfn: use GEOSRelatePattern for ST_ContainsProperly

GEOSRelatePattern seems to avoid the checks for wildcards, making
ST_ContainsProperly faster.

Release note: None

otan commented:
bors r=sumeerbhola
